olive/data/component/pre_process_data.py

```diff
@@ -128,121 +137,458 @@ def _tokenizer_and_align_labels(examples):
+    def _check_padding(cls, v, values):
+        if "pad_to_max_len" not in values:
+            ValueError("Invalid pad_to_max_len")
+        if v and values["pad_to_max_len"]:
+            raise ValueError("pad_to_max_len and drop_short_sequences cannot both be True")
+        return v
+
+
+class TextGenCorpusParams(TextGenParams):
+    """Parameters for text generation task with 'corpus' dataset type."""
+
+    text_cols: list  # list of text columns
+    corpus_strategy: TextGenCorpusStrategy = TextGenCorpusStrategy.LINE_BY_LINE
+    stride: int = None  # required when corpus_strategy is JOIN_SLIDING_WINDOW
+    joiner: str = " "  # delimiter to use when joining the rows of the input columns.
+    random_seed: int = None  # random seed for LINE_BY_LINE_RANDOM and JOIN_RANDOM
+    random_retries: int = (
+        10  # number of resamples to try before giving up when a sample is too short for RANDOM strategies
+    )
+
+    @validator("stride", always=True)
+    def _check_stride(cls, v, values):
+        if "corpus_strategy" not in values:
+            raise ValueError("Invalid corpus_strategy")
+        if values["corpus_strategy"] == TextGenCorpusStrategy.JOIN_SLIDING_WINDOW and v is None:
+            raise ValueError("stride must be specified when corpus_strategy is JOIN_SLIDING_WINDOW")
+        return v
+
+    @validator("corpus_strategy", always=True)
+    def _check_max_samples(cls, v, values):
+        if "max_samples" not in values:
+            raise ValueError("Invalid max_samples")
+        if "random" in v and values["max_samples"] is None:
+            raise ValueError("max_samples must be specified when corpus_strategy is random")
+        return v
+
+    @validator("random_seed", always=True)
+    def _check_random(cls, v, values):
+        if "corpus_strategy" not in values:
+            raise ValueError("Invalid corpus_strategy")
+        if "random" in values["corpus_strategy"] and v is None:
+            raise ValueError("random_seed must be specified when corpus_strategy is random")
+        return v
+
+
+class TextGenPairParams(TextGenParams):
+    """Parameters for text generation task with 'pair' dataset type."""
+
+    pair_format: TextGenPairFormat = TextGenPairFormat.DEFAULT
+    input_col: str = None  # required when pair_format is CUSTOM
+    output_col: str = None  # required when pair_format is CUSTOM
+    target_max_len: int  # max length of target sequence
+    ignore_source_in_labels: bool = False  # set source tokens to ignore_index in labels
+
+    @validator("input_col", "output_col", always=True)
+    def _check_custom(cls, v, field, values):
+        if "pair_format" not in values:
+            raise ValueError("Invalid pair_format")
+        if values["pair_format"] == TextGenPairFormat.CUSTOM and v is None:
+            raise ValueError(f"{field.name} must be specified when pair_format is CUSTOM")
+        return v
+
+
 @Registry.register_pre_process()
 def text_generation_huggingface_pre_process(
```

Comment:
Could we move those Params class and help function to a separated module? Like componnet.generation or something else. 
It is better to keep pre_process_data.py clear then user/dev can know which kind of function is used for data_config.

Reply:
Yes, I can move them to a separate file. I was also thinking that this file is going to become unwieldy as we add more tasks. But wasn't sure if I should move the preprocessing func also. 

I will only move the helper classes and functions as you suggested.

Reply:
Done!